### DISCUSSION TITLE
RFC: Advanced playlists using clocks

### DISCUSSION THREAD

**Discussion by bjarn** [AzuraCast/AzuraCast#5407] [Ideas]
### Problem
Currently, on the 1.0 roadmap, you can find the following item:

> Grouped/Nested Playlists: One of the most widely-requested features that we don't currently support is the idea of "nested" playlists; one overarching playlist that contains other ones in a specified playback order. For example, one parent playlist could say "Play X songs from playlist 1, then Y songs from playlist 2, then Y from playlist 3". Although our code doesn't currently support this, implementing it is, at least in my opinion, imperative for a 1.0 release.

While the current playlists work fine, they are pretty basic and tough to set up with weights for more professional scheduling.

### Solution
Professional radio stations work with so-called _Clocks_. My proposal is to introduce a similar system like this in AzuraCast and work-out this idea together.
The idea of having Clocks is pretty similar to the idea of grouped/nested playlists.

In the navigation, you will find `Clocks`, `Playlists` and `Schedule`. Where playlists' functionality will stay mostly the same, clocks and schedule is new. The old schedule functionality will be moved.

#### Clocks
Creating a clock provides a new UI with drag and drop functionality, showing the existing playlists. A clock can't be larger than 1 hour, or it will be an anti-pattern.
You can drag and reorder playlists into the clock, telling it how long the current playlist should be used. 

An example of a broadcasting clock to give an idea what it is in case people don't know:
![image](https://user-images.githubusercontent.com/14638441/169774683-201bc980-43bb-4a05-93d7-e7bd2f626ceb.png)

#### Schedule
Next, in the actual schedule view, you can drag clocks and tell them how long they should play. A clock can be repeated multiple times.

If you just want a playlist, it is required to create a clock with just a single playlist. It might be convenient to auto-create a new clock for just that playlist when dragging a playlist into the schedule view.

#### Playlists
Playlists weights will be obsolete and only required when using manual auto DJ (aka Liquidsoap).

### Why clocks?
Broadcast clocks are an industry standard in radio broadcasting. In order to professionalise internet radio and AzuraCast, this might be a very cool feature to introduce.

Clocks also allow you to set TOTH and BOTM (Top Of The Hour & Bottom Of The Hour), ad breaks and other specific items more easily.

**Comment by amarildoazura** [comment]
Very good request is a demand of online radios in Brazil the question of the right time, because by default the radios use the automatic locution of the right time.

**Comment by cpaschen** [comment]
This is THE one feature that Azuracast really lacks to be adopted by those of us who have ever worked in terrestrial radio. 
Creating formatics using the current playlist just doesn't give you the consistent format of having the ability to create a 'clock' and apply the clock to 'day parts'.
(Although I've finally just decided to migrate to Azuracast from my previous system, and loose consistency, because the other system was no longer stable or sustainable.)
Having clock functionality (or even the Grouped/Nested playlists) would be a huge welcome addition.
